sklearn/metrics/base.py

```diff
@@ -131,3 +132,69 @@ def _average_binary_score(binary_metric, y_true, y_score, average,
         return np.average(score, weights=average_weight)
     else:
         return score
+
+
+def _average_multiclass_ovo_score(binary_metric, y_true, y_score, average):
+    """Uses the binary metric for one-vs-one multiclass classification,
+    where the score is computed according to the Hand & Till (2001) algorithm.
+
+    Parameters
+    ----------
+    y_true : array, shape = [n_samples]
+        True multiclass labels.
+        Assumes labels have been recoded to 0 to n_classes.
+
+    y_score : array, shape = [n_samples, n_classes]
+        Target scores corresponding to probability estimates of a sample
+        belonging to a particular class
+
+    average : 'macro' or 'weighted', default='macro'
+        ``'macro'``:
+            Calculate metrics for each label, and find their unweighted
+            mean. This does not take label imbalance into account. Classes
+            are assumed to be uniformly distributed.
+        ``'weighted'``:
+            Calculate metrics for each label, taking into account the a priori
+            distribution of the classes.
+
+    binary_metric : callable, the binary metric function to use.
+        Accepts the following as input
+            y_true_target : array, shape = [n_samples_target]
+                Some sub-array of y_true for a pair of classes designated
+                positive and negative in the one-vs-one scheme.
+            y_score_target : array, shape = [n_samples_target]
+                Scores corresponding to the probability estimates
+                of a sample belonging to the designated positive class label
+
+    Returns
+    -------
+    score : float
+        Average the sum of pairwise binary metric scores
+    """
+    n_labels = len(np.unique(y_true))
+    pos_and_neg_prevalence = []
+    label_scores = []
+    for pos, neg in itertools.combinations(range(n_labels), 2):
+        pos_ix = y_true == pos
+        ix = np.logical_or(pos_ix, y_true == neg)
+
+        pos_and_neg_prevalence.append(float(np.sum(ix)) / len(y_true))
+
+        y_score_filtered = y_score[ix]
+
+        class_a = pos_ix[ix]
+        class_b = np.logical_not(class_a)
+
+        score_class_a = binary_metric(
+                class_a, y_score_filtered[:, pos])
+        score_class_b = binary_metric(
+                class_b, y_score_filtered[:, neg])
+        binary_avg_score = (score_class_a + score_class_b) / 2.
+        label_scores.append(binary_avg_score)
+
+    if average == "weighted":
```

Comment:
Wanted to ask about this: my current test case passes, but I think my understanding of weighted ovo may just be incorrect. I am looking at [this case (page 4, AU1P)](https://www.math.ucdavis.edu/~saito/data/roc/ferri-class-perf-metrics.pdf) when I am implementing weighted ovo; however, I'm getting values that are below .5 for the test cases using this.